astropy/tests/pytest_doctestplus.py

```diff
@@ -0,0 +1,361 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from pkgutil import find_loader
+# This is necessary to prevent pytest from reading the doctestplus plugin when
+# the pytest_doctestplus standalone package is installed
+if find_loader('pytest_doctestplus') is None:
+    from ..extern import six
+
+    import doctest
+    import fnmatch
+    import imp
+    import os
+    import re
+    import sys
+
+    import pytest
+
+    from .output_checker import AstropyOutputChecker, FIX
+
+    try:
+        import importlib.machinery as importlib_machinery
+    except ImportError:  # Python 2.7
+        importlib_machinery = None
+
+    # these pytest hooks allow us to mark tests and run the marked tests with
+    # specific command line options.
+
+
+    def pytest_addoption(parser):
+        parser.addoption("--doctest-plus", action="store_true",
+                         help="enable running doctests with additional "
+                         "features not found in the normal doctest "
+                         "plugin")
+
+        parser.addoption("--doctest-rst", action="store_true",
+                         help="enable running doctests in .rst documentation")
+
+        parser.addini("doctest_plus", "enable running doctests with additional "
+                      "features not found in the normal doctest plugin")
+
+        parser.addini("doctest_norecursedirs",
+                      "like the norecursedirs option but applies only to doctest "
+                      "collection", type="args", default=())
+
+        parser.addini("doctest_rst",
+                      "Run the doctests in the rst documentation",
+                      default=False)
+
+
+    # We monkey-patch in our replacement doctest OutputChecker.  Not
+    # great, but there isn't really an API to replace the checker when
+    # using doctest.testfile, unfortunately.
+    doctest.OutputChecker = AstropyOutputChecker
+
+    REMOTE_DATA = doctest.register_optionflag('REMOTE_DATA')
+
+
+    def pytest_configure(config):
+
+        doctest_plugin = config.pluginmanager.getplugin('doctest')
+        if (doctest_plugin is None or config.option.doctestmodules or not
+                (config.getini('doctest_plus') or config.option.doctest_plus)):
+            return
+
+        # These are the default doctest options we use for everything.
+        # There shouldn't be any need to manually put them in doctests
+        # themselves.
+        opts = (doctest.ELLIPSIS |
+                doctest.NORMALIZE_WHITESPACE |
+                FIX)
+
+        class DocTestModulePlus(doctest_plugin.DoctestModule):
+            # pytest 2.4.0 defines "collect".  Prior to that, it defined
```

Comment:
is this collect() below then just needed as a workaround for pytest <2.4? We already require 2.8 to the 2.0.x branch, so if yes, please remove.